PocketKit/Sources/PocketKit/Collections/CollectionViewController.swift

```diff
@@ -318,6 +289,123 @@ private extension CollectionViewController {
         }
     }
 
+    /// Determines the stories layout in the native collection view on iPad mode with regular horizontal size class (including split view). Number of columns for the grid layout depends on device orientation and the view's width length. If view's width is less than 800, show 2 col, otherwise show 2 col if it is in portrait and 3 col if it is in landscape mode.
+    private func collectionSectionForWideLayout(with stories: [CollectionStory], width: CGFloat, margin: CGFloat) -> NSCollectionLayoutSection {
+        if self.view.bounds.width >= Constants.minWidthBoundaryForFullColumnLayout && !UIDevice.current.orientation.isPortrait {
+            return collectionSectionForiPadLayout(with: stories, width: width, margin: margin, numberOfColumns: 3)
+        } else {
+            return collectionSectionForiPadLayout(with: stories, width: width, margin: margin, numberOfColumns: 2)
+        }
+    }
```

Comment:
```suggestion
    private func collectionSectionForWideLayout(with stories: [CollectionStory], width: CGFloat, margin: CGFloat) -> NSCollectionLayoutSection {
        collectionSectionForiPadLayout(with: stories, width: width, margin: margin, numberOfColumns: numberOfColumns())
    }
    
    private func numberOfColumns() -> Int {
        if self.view.bounds.width >= Constants.minWidthBoundaryForFullColumnLayout && !UIDevice.current.orientation.isPortrait {
            return 2
         } else {
            return 3
         }
    }
```
Break up the logic a little.

Reply:
I think the function `collectionSectionForWideLayout` is meaningless here now, so instead I will do the following:

```
    private func slateSectionForWideLayout(with recommendations: [Recommendation], width: CGFloat, sideMargin: CGFloat) -> NSCollectionLayoutSection {
        let numberOfColumns = showThreeColumnLayout ? 3 : 2
        return slateSectionForiPadLayout(with: recommendations, width: width, sideMargin: sideMargin, numberOfColumns: numberOfColumns)
    }
```

let me know if this works!